camunda-commons-ui/lib/widgets/password/cam-widget-password.js

```diff
@@ -81,49 +69,68 @@ module.exports = [
+          $scope.$watch(
+            'profile',
+            value => {
+              if (value) {
+                profile = value;
+                handlePasswordUpdate();
+              }
+            },
+            true
+          );
+        });
+
         // Wait a second for more user input before validating
         var createRestCall = debounce(function() {
           if (!$scope.password) {
             return;
           }
 
           passwordPolicyProvider
-            .validate($scope.password, function(err, res) {
-              if (err) {
-                $scope.loadingState = 'NOT_OK';
-                $scope.tooltipText = $sce.trustAsHtml(
-                  $translate.instant('PASSWORD_POLICY_TOOLTIP_ERROR')
-                );
-                return;
-              }
-              if (res.valid) {
-                $scope.loadingState = 'OK';
-                $scope.isValid = true;
-              } else {
-                $scope.loadingState = 'NOT_OK';
-                $scope.isValid = false;
-
-                $scope.tooltipText = $translate.instant(
-                  'PASSWORD_POLICY_TOOLTIP_PARTIAL'
-                );
-                $scope.tooltipText += '<ul>';
-                res.rules.forEach(function(rule) {
-                  if (!rule.valid) {
-                    $scope.tooltipText +=
-                      '<li>' +
-                      $translate.instant(rule.placeholder, rule.parameter) +
-                      '</li>';
-                  }
-                });
-                $scope.tooltipText += '</ul>';
-
-                $scope.tooltipText = $sce.trustAsHtml($scope.tooltipText);
+            .validate(
+              {
+                password: $scope.password,
+                profile: {
+                  id: sanitize(profile.id),
+                  firstName: sanitize(profile.firstName),
+                  lastName: sanitize(profile.lastName),
+                  email: sanitize(profile.email)
+                }
+              },
+              function(err, res) {
+                if (err) {
+                  $scope.loadingState = 'NOT_OK';
+                  $scope.tooltip = 'PASSWORD_POLICY_TOOLTIP_ERROR';
+                  return;
+                }
+                if (res.valid) {
+                  $scope.loadingState = 'OK';
+                  $scope.isValid = true;
+                } else {
+                  $scope.loadingState = 'NOT_OK';
+                  $scope.isValid = false;
+
+                  $scope.tooltip = 'PASSWORD_POLICY_TOOLTIP_PARTIAL';
+
+                  $scope.rules = res.rules;
+                }
               }
-            })
+            )
             .catch(angular.noop);
         }, 1000);
 
         $scope.$watch('[password]', handlePasswordUpdate, true);
+
+        function sanitize(value) {
+          if (value === undefined) {
+            return '';
+          }
+
+          return value === null ? '' : value;
+        }
```

Comment:
Let's move this function somewhere before `passwordPolicyProvider.validate`. When reading the code, we then see it defined before we use it.

undefined and null are both false-y values, so we can shorten this function to 
```suggestion
        function sanitize(value) {
          return value || '';
        }
```

or, for added style points:
```suggestion
        var sanitize = value => value || '';
```

Reply:
👍